diff --git a/gopls/pkg/lsp/protocol/tsclient.go b/gopls/pkg/lsp/protocol/tsclient.go
index b308950..6fa1a3a 100644
--- a/gopls/pkg/lsp/protocol/tsclient.go
+++ b/gopls/pkg/lsp/protocol/tsclient.go
@@ -14,7 +14,6 @@ import (
 	"context"
 	"encoding/json"
 
-	"github.com/kralicky/tools-lite/gopls/pkg/util/bug"
 	"github.com/kralicky/tools-lite/pkg/jsonrpc2"
 )
 
@@ -44,7 +43,6 @@ type Client interface {
 func clientDispatch(ctx context.Context, client Client, reply jsonrpc2.Replier, r jsonrpc2.Request) (bool, error) {
 	defer func() {
 		if x := recover(); x != nil {
-			bug.Reportf("client panic in %s request", r.Method())
 			panic(x)
 		}
 	}()
diff --git a/gopls/pkg/lsp/protocol/tsserver.go b/gopls/pkg/lsp/protocol/tsserver.go
index c53ec90..6438f06 100644
--- a/gopls/pkg/lsp/protocol/tsserver.go
+++ b/gopls/pkg/lsp/protocol/tsserver.go
@@ -14,7 +14,6 @@ import (
 	"context"
 	"encoding/json"
 
-	"github.com/kralicky/tools-lite/gopls/pkg/util/bug"
 	"github.com/kralicky/tools-lite/pkg/jsonrpc2"
 )
 
@@ -98,7 +97,6 @@ type Server interface {
 func serverDispatch(ctx context.Context, server Server, reply jsonrpc2.Replier, r jsonrpc2.Request) (bool, error) {
 	defer func() {
 		if x := recover(); x != nil {
-			bug.Reportf("server panic in %s request", r.Method())
 			panic(x)
 		}
 	}()
