diff --git a/gopls/internal/protocol/mapper.go b/gopls/internal/protocol/mapper.go
index 85997c24d..d1bd957a9 100644
--- a/gopls/internal/protocol/mapper.go
+++ b/gopls/internal/protocol/mapper.go
@@ -71,7 +71,6 @@ import (
 	"sync"
 	"unicode/utf8"

-	"github.com/kralicky/tools-lite/gopls/internal/util/bug"
 	"github.com/kralicky/tools-lite/gopls/internal/util/safetoken"
 )

@@ -132,14 +131,6 @@ func (m *Mapper) initLines() {
 // LineCol8Position converts a valid line and UTF-8 column number,
 // both 1-based, to a protocol (UTF-16) position.
 func (m *Mapper) LineCol8Position(line, col8 int) (Position, error) {
-	// Report a bug for inputs that are invalid for any file content.
-	if line < 1 {
-		return Position{}, bug.Errorf("invalid 1-based line number: %d", line)
-	}
-	if col8 < 1 {
-		return Position{}, bug.Errorf("invalid 1-based column number: %d", col8)
-	}
-
 	m.initLines()
 	line0 := line - 1 // 0-based
 	if !(0 <= line0 && line0 < len(m.lineStart)) {
