diff --git a/gopls/internal/protocol/generate/tables.go b/gopls/internal/protocol/generate/tables.go
index 5182bc2..6bc437a 100644
--- a/gopls/internal/protocol/generate/tables.go
+++ b/gopls/internal/protocol/generate/tables.go
@@ -62,7 +62,6 @@ var usedGoplsStar = make(map[prop]bool)
 var renameProp = map[prop]string{
 	{"CancelParams", "id"}:         "interface{}",
 	{"Command", "arguments"}:       "[]json.RawMessage",
-	{"CompletionItem", "textEdit"}: "TextEdit",
 	{"CodeAction", "data"}:         "json.RawMessage", // delay unmarshalling commands
 	{"Diagnostic", "code"}:         "interface{}",
 	{"Diagnostic", "data"}:         "json.RawMessage", // delay unmarshalling quickfixes
@@ -123,8 +122,6 @@ var usedDisambiguate = make(map[string]bool)
 var goplsType = map[string]string{
 	"And_RegOpt_textDocument_colorPresentation": "WorkDoneProgressOptionsAndTextDocumentRegistrationOptions",
 	"ConfigurationParams":                       "ParamConfiguration",
-	"DocumentDiagnosticParams":                  "string",
-	"DocumentDiagnosticReport":                  "string",
 	"DocumentUri":                               "DocumentURI",
 	"InitializeParams":                          "ParamInitialize",
 	"LSPAny":                                    "interface{}",
